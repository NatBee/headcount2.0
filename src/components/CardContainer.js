import React from 'react';
import Card from './Card';
import '../styles/CardContainer.css';
import PropTypes from 'prop-types';


const CardContainer = (props) => {

    const dataArray = Object.keys(props.allDistrictData);
    const newCard = dataArray.map((location, index) => <Card data={ props.allDistrictData[location]} key={index} handleClick={props.compareCards}/>);
  
    return (
        <div className='card-section'>
            <h2 className='cards-title'>School Data by District</h2>
            <section className='card-wrap'>
                { newCard }
            </section>
        </div>
    );
};

CardContainer.propTypes = {
    compareCards: PropTypes.func,
    allDistrictData: PropTypes.array,
    props: PropTypes.shape({
        data: PropTypes.Object
    })
}

export default CardContainer;