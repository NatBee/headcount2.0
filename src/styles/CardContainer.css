.container-wrap {
  background-color: #535C78;
  display: flex;
  flex-wrap: wrap;
  height: 400px;
  justify-content: space-evenly;
  overflow: scroll;
  width: 100%;
}

