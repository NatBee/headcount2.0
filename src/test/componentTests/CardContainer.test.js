import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from '../../components/CardContainer';
import { shallow, mount } from 'enzyme';

const mockProps = {
  data: {
    data:{  
      2004:1,
      2005:1,
      2006:1,
      2007:1,
      2008:1,
      2009:1,
      2010:1,
      2011:1,
      2012:1,
      2013:1,
      2014:1
    },
    location: 'COLORADO'}  
  };

it('should match the snapshot', () => {
  const renderedComponent = shallow(<CardContainer allDistrictData={mockProps} />)
  expect(renderedComponent).toMatchSnapshot()
})

it('should render expected number of cards', () => {
  
  const renderedComponent = mount(<CardContainer allDistrictData={mockProps} />)
  expect(renderedComponent.find('.card').length).toEqual(1);
})
