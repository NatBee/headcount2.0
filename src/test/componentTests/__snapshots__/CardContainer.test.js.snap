// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    allDistrictData={
        Object {
            "data": Object {
              "data": Object {
                "2004": 1,
                "2005": 1,
                "2006": 1,
                "2007": 1,
                "2008": 1,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "location": "COLORADO",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="cards-title"
>
          School Data by District
</h2>,
        <section
          className="card-wrap"
>
          <Card
                    data={
                              Object {
                                        "data": Object {
                                          "2004": 1,
                                          "2005": 1,
                                          "2006": 1,
                                          "2007": 1,
                                          "2008": 1,
                                          "2009": 1,
                                          "2010": 1,
                                          "2011": 1,
                                          "2012": 1,
                                          "2013": 1,
                                          "2014": 1,
                                        },
                                        "location": "COLORADO",
                                      }
                    }
                    handleClick={undefined}
          />
</section>,
      ],
      "className": "card-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "School Data by District",
          "className": "cards-title",
        },
        "ref": null,
        "rendered": "School Data by District",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              data={
                            Object {
                                          "data": Object {
                                            "2004": 1,
                                            "2005": 1,
                                            "2006": 1,
                                            "2007": 1,
                                            "2008": 1,
                                            "2009": 1,
                                            "2010": 1,
                                            "2011": 1,
                                            "2012": 1,
                                            "2013": 1,
                                            "2014": 1,
                                          },
                                          "location": "COLORADO",
                                        }
              }
              handleClick={undefined}
/>,
          ],
          "className": "card-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "data": Object {
                  "2004": 1,
                  "2005": 1,
                  "2006": 1,
                  "2007": 1,
                  "2008": 1,
                  "2009": 1,
                  "2010": 1,
                  "2011": 1,
                  "2012": 1,
                  "2013": 1,
                  "2014": 1,
                },
                "location": "COLORADO",
              },
              "handleClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="cards-title"
>
            School Data by District
</h2>,
          <section
            className="card-wrap"
>
            <Card
                        data={
                                    Object {
                                                "data": Object {
                                                  "2004": 1,
                                                  "2005": 1,
                                                  "2006": 1,
                                                  "2007": 1,
                                                  "2008": 1,
                                                  "2009": 1,
                                                  "2010": 1,
                                                  "2011": 1,
                                                  "2012": 1,
                                                  "2013": 1,
                                                  "2014": 1,
                                                },
                                                "location": "COLORADO",
                                              }
                        }
                        handleClick={undefined}
            />
</section>,
        ],
        "className": "card-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "School Data by District",
            "className": "cards-title",
          },
          "ref": null,
          "rendered": "School Data by District",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                data={
                                Object {
                                                "data": Object {
                                                  "2004": 1,
                                                  "2005": 1,
                                                  "2006": 1,
                                                  "2007": 1,
                                                  "2008": 1,
                                                  "2009": 1,
                                                  "2010": 1,
                                                  "2011": 1,
                                                  "2012": 1,
                                                  "2013": 1,
                                                  "2014": 1,
                                                },
                                                "location": "COLORADO",
                                              }
                }
                handleClick={undefined}
/>,
            ],
            "className": "card-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "data": Object {
                  "data": Object {
                    "2004": 1,
                    "2005": 1,
                    "2006": 1,
                    "2007": 1,
                    "2008": 1,
                    "2009": 1,
                    "2010": 1,
                    "2011": 1,
                    "2012": 1,
                    "2013": 1,
                    "2014": 1,
                  },
                  "location": "COLORADO",
                },
                "handleClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
